### PR TITLE
MueLu: Cleanup up matrix dumping

### PR DESCRIPTION
**Pull request by csiefer2**
Issue: #11214
Tests: n/a
Stakeholder feedback: n/a

### CODE REVIEW

**Review by GrahamBenHarper — COMMENTED** [review]
Is this covered by existing tests?

**Review by GrahamBenHarper — DISMISSED** [review]
Good enough for me

File: packages/muelu/doc/UsersGuide/masterList.xml [hunk 1]
```diff
@@ -1030,6 +1030,16 @@
       <comment-ML>not supported by ML</comment-ML>
     </parameter>
 
+    <parameter>
+      <name>save data</name>
+      <type>string</type>
+      <default>"{}"</default>
+      <description>Data to dump to disk after the
+      hierarchy is constructed.  Use comma-seperated array in braces</description>
```
Comment (jhux2) [inline packages/muelu/doc/UsersGuide/masterList.xml:1038]: "separated"

File: packages/muelu/doc/UsersGuide/paramlist_hidden.tex [hunk 2]
```diff
@@ -232,6 +232,9 @@
 \cbb{keep data}{string}{"{}"}{Keep a subset of the hierarchy data after the
       hierarchy is constructed.  Use comma-seperated array in braces}
         
+\cbb{save data}{string}{"{}"}{Data to dump to disk after the
+      hierarchy is constructed.  Use comma-seperated array in braces}
```
Comment (jhux2) [inline packages/muelu/doc/UsersGuide/paramlist_hidden.tex:236]: ditto

File: packages/muelu/src/Interface/MueLu_HierarchyManager.hpp [hunk 3]
```diff
@@ -364,14 +375,17 @@ namespace MueLu {
...
-    Teuchos::Array<int> prolongatorsToPrint_;
-    Teuchos::Array<int> restrictorsToPrint_;
+    // Items here get handled manually
     Teuchos::Array<int> nullspaceToPrint_;
     Teuchos::Array<int> coordinatesToPrint_;
     Teuchos::Array<int> aggregatesToPrint_;
     Teuchos::Array<int> elementToNodeMapsToPrint_;
-    Teuchos::Array<std::string> dataToSave_;
+
+    // Data we'll need to save, not neccesarily print
```
Comment (jhux2) [inline packages/muelu/src/Interface/MueLu_HierarchyManager.hpp:384]: "necessarily"

**Review by jhux2 — APPROVED** [review]
Nice!  There are a few typos in some spots.

**Review by GrahamBenHarper — APPROVED** [review]

**Pull request merged**